gemini/reasoning-engine/tutorial_langgraph.ipynb

```diff
@@ -0,0 +1,665 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "id": "NyKGtVQjgx13"
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "if \"google.colab\" in sys.modules:\n",
+    "    from google.colab import auth\n",
+    "\n",
+    "    auth.authenticate_user()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "DF4l8DTdWgPY"
+   },
+   "source": [
+    "### Set Google Cloud project information and initialize Vertex AI SDK\n",
+    "\n",
+    "To get started using Vertex AI, you must have an existing Google Cloud project and [enable the Vertex AI API](https://console.cloud.google.com/flows/enableapi?apiid=aiplatform.googleapis.com).\n",
+    "\n",
+    "Learn more about [setting up a project and a development environment](https://cloud.google.com/vertex-ai/docs/start/cloud-environment)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "Nqwi-5ufWp_B"
+   },
+   "outputs": [],
+   "source": [
+    "# PROJECT_ID = \"[your-project-id]\"  # @param {type:\"string\"}\n",
+    "PROJECT_ID = \"koverholt-devrel-355716\"  # @param {type:\"string\"}\n",
+    "LOCATION = \"us-central1\"  # @param {type:\"string\"}\n",
+    "# STAGING_BUCKET = \"gs://[your-staging-bucket]\"  # @param {type:\"string\"}\n",
+    "STAGING_BUCKET = \"gs://koverholt-devrel-355716\"  # @param {type:\"string\"}\n",
+    "\n",
+    "import vertexai\n",
+    "\n",
+    "vertexai.init(project=PROJECT_ID, location=LOCATION, staging_bucket=STAGING_BUCKET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EdvJRUWRNGHE"
+   },
+   "source": [
+    "## Building and deploying a LangGraph app on Reasoning Engine\n",
+    "\n",
+    "In the following sections, we'll walk through the process of building and deploying a LangGraph application using Reasoning Engine in Vertex AI."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import libraries\n",
+    "\n",
+    "Import the necessary Python libraries. These libraries provide the tools we need to interact with LangGraph, Vertex AI, and other components of our application."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from vertexai.preview import reasoning_engines\n",
+    "from langchain_google_vertexai import ChatVertexAI\n",
+    "from langchain_core.messages import HumanMessage, BaseMessage\n",
+    "from langgraph.graph import END, MessageGraph\n",
+    "from langchain_core.tools import tool\n",
+    "from langgraph.prebuilt import ToolNode\n",
+    "from typing import Literal, List"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define tools\n",
+    "\n",
+    "You'll start by defining the a tool for your LangGraph application. You'll define a custom Python function that act as tools in our agentic application.\n",
```

Comment:
This might be extra of me, but I would like if `tool` was a hyperlink to the LangGraph docs for that feature. Same with other features of LangGraph

Reply:
Good callout, I added links to the respective LangChain components and conceptual docs. Thanks!